pkg/maintainer/spv/spv.go

```diff
@@ -57,10 +98,302 @@ func (sm *spvMaintainer) startControlLoop(ctx context.Context) {
 }
 
 func (sm *spvMaintainer) maintainSpv(ctx context.Context) error {
```

Comment:
I really like how the control flow is organized in those functions. The `for` loops, the `select`s, and how the functions are divided. It makes a really logical structure that is easy to follow. 👌🏻